File: bloom.py
d={0:0,1:0,2:0,3:0,4:0,5:0,6:0}

def h1(e):
    return (2*e+1)%7

def h2(e):
    return (5*e+3)%7

def h3(e):
    return (9*e+6)%7

def search(e):
    a=d[h1(e)]
    b=d[h2(e)]
    c=d[h3(e)]
    print("hash values are : ",(h1(e),h2(e),h3(e)))
    print("the bloom filter values at these hashes are : ",(a,b,c))
    if a==1 and b==1 and c==1:
        print("ALREADY EXISTS\n")
    else:
        d[h1(e)]=1
        d[h2(e)]=1
        d[h3(e)]=1
        print("INSERTED\n")

choice=0
while True:
    if choice==0:
        e=int(input("enter element to insert : "))
        search(e)
        choice=-1
    elif choice==-1:
        print("******************* 0=search and insert\t 1=exit\t 2=print************************")
        choice=int(input("enter choice : "))
    elif choice==1:
        break
    elif choice==2:
        print(d)
        choice=-1

File: prac/bloom.java
import java.io.*;
import java.util.*;

public class bloom{
    public static int h1(int ele){
        return (5*ele+2)%7;
    }
    public static int h2(int ele){
        return (89*ele+27)%7;
    }
    public static void main(String args[]){
        Scanner sc =new Scanner(System.in);
        Hashtable<Integer, Integer> ht = new Hashtable<>(7);
        for(int i=0;i<7;i++){
            ht.put(i,0);
            }
        int choice=1;
        while(true){
            if(choice==1)
            {
                System.out.print("Enter element to insert : ");
                int ele=sc.nextInt();
                int a=h1(ele);
                int b=h2(ele);
                System.out.println("hash values are : "+a+ " ," +b);
                
                if((ht.get(a)==1) && (ht.get(b)==1)){
                    System.out.println("element already exists \n");
                }
                else{
                    ht.put(a,1);
                    ht.put(b,1);
                    System.out.println("element inserted \n");
                }
                choice=-1;
                
                
            }
            else if(choice==2){
                System.out.println(ht);
                choice=-1;
            }
            else if(choice ==-1){
                System.out.println("*****1==insert or search\t 2==print ht\t 3==exit*****");
                System.out.print("enter choice : ");
                choice=sc.nextInt();
            }
            else if(choice==3){
                break ;
            }
            
        }
        
    }

}